sdk/src/azure/platform/az_pahoasync.c

```diff
@@ -0,0 +1,619 @@
+
+  return _az_result_from_pahoasync(
+      MQTTProperties_add(property_bag->pahoasync_properties, &(property_bag->pahoasync_property)));
+}
+
+AZ_NODISCARD az_result az_mqtt5_property_bag_append_byte(
+    az_mqtt5_property_bag* property_bag,
+    az_mqtt5_property_type type,
+    uint8_t prop_byte)
+{
+  _az_PRECONDITION_NOT_NULL(property_bag);
+
+  property_bag->pahoasync_property.identifier = (enum MQTTPropertyCodes)type;
+  property_bag->pahoasync_property.value.byte = prop_byte;
+
+  return _az_result_from_pahoasync(
+      MQTTProperties_add(property_bag->pahoasync_properties, &(property_bag->pahoasync_property)));
+}
+
+AZ_NODISCARD az_result az_mqtt5_property_bag_append_int(
+    az_mqtt5_property_bag* property_bag,
+    az_mqtt5_property_type type,
+    uint32_t prop_int)
+{
+  _az_PRECONDITION_NOT_NULL(property_bag);
+
+  property_bag->pahoasync_property.identifier = (enum MQTTPropertyCodes)type;
+  property_bag->pahoasync_property.value.integer4 = prop_int; // TODO_L: Add append for two byte int
+
+  return _az_result_from_pahoasync(
+      MQTTProperties_add(property_bag->pahoasync_properties, &(property_bag->pahoasync_property)));
+}
+
+AZ_NODISCARD az_result az_mqtt5_property_bag_append_binary(
+    az_mqtt5_property_bag* property_bag,
+    az_mqtt5_property_type type,
+    az_mqtt5_property_binarydata* prop_bindata)
+{
+  _az_PRECONDITION_NOT_NULL(property_bag);
+  _az_PRECONDITION_NOT_NULL(prop_bindata);
+
+  property_bag->pahoasync_property.identifier = (enum MQTTPropertyCodes)type;
+  property_bag->pahoasync_property.value.data.data = (char*)az_span_ptr(prop_bindata->bindata);
+  property_bag->pahoasync_property.value.data.len = az_span_size(prop_bindata->bindata);
+
+  return _az_result_from_pahoasync(
+      MQTTProperties_add(property_bag->pahoasync_properties, &(property_bag->pahoasync_property)));
+}
+
+AZ_NODISCARD az_result az_mqtt5_property_bag_read_string(
+    az_mqtt5_property_bag* property_bag,
+    az_mqtt5_property_type type,
+    az_mqtt5_property_string* out_prop_str)
+{
+  _az_PRECONDITION_NOT_NULL(property_bag);
+  _az_PRECONDITION_NOT_NULL(out_prop_str);
+  MQTTProperty* prop;
+
+  prop = MQTTProperties_getProperty(
+      property_bag->pahoasync_properties, (enum MQTTPropertyCodes)type);
+
+  if (prop == NULL)
+  {
+    return AZ_ERROR_ITEM_NOT_FOUND;
+  }
+
+  out_prop_str->str = az_span_create((uint8_t*)prop->value.data.data, prop->value.data.len);
+
+  return AZ_OK;
+}
+
+AZ_NODISCARD az_result az_mqtt5_property_bag_find_stringpair(
+    az_mqtt5_property_bag* property_bag,
+    az_mqtt5_property_type type,
+    az_span key,
+    az_mqtt5_property_stringpair* out_prop_strpair)
+{
+  _az_PRECONDITION_NOT_NULL(property_bag);
+  _az_PRECONDITION_NOT_NULL(out_prop_strpair);
+  MQTTProperty* prop;
+  int stringpair_index = 0;
+
+  prop = MQTTProperties_getPropertyAt(
+      property_bag->pahoasync_properties, (enum MQTTPropertyCodes)type, stringpair_index);
+
+  while (prop != NULL)
```

Comment:
```suggestion
  for (prop =  MQTTProperties_getPropertyAt(property_bag->pahoasync_properties, (enum MQTTPropertyCodes)type, stringpair_index); prop != NULL; prop =  MQTTProperties_getPropertyAt(property_bag->pahoasync_properties, (enum MQTTPropertyCodes)type, stringpair_index))
```

not sure if something like this makes things more concise about switching to the next prop or not

Reply:
Switched it to your approach, it makes it more concise :)